modules/apps/asset/asset-categories-admin-web/src/main/resources/META-INF/resources/category/properties.jsp

```diff
@@ -146,4 +146,6 @@ renderResponse.setTitle(category.getTitle(locale));
 	if (categoryPropertiesTrigger) {
 		categoryPropertiesTrigger.setData('autoFieldsInstance', autoFields);
 	}
-</aui:script>
\ No newline at end of file
+</aui:script>
+
+<liferay-frontend:stylesheet bundle="com.liferay.frontend.css.shared" css="lfr_form_row.css" />
```

Comment:
Is there anything in common to all these jsps that we could use to automate this? Are they all using `aui:form` or some other tag where we could centralize the load of `lfr_form_row`? That way it wouldn't be up to each developer to manually add the css.

Reply:
wouldn't it lose the main advantage of this taglib (load the CSS only where it's necessary)? We could add it to the `init.jsp` when it's loaded several times in the same module, but it seems a replica of the issue we have with the current theme system (in a smaller scale)

Reply:
> wouldn't it lose the main advantage of this taglib (load the CSS only where it's necessary)? 

Not necessarily. That can be achieved in many different ways. This taglib is just wrapping some of those mechanisms for usage within JSPs, but that doesn't mean that's the only means to achieve it.

> We could add it to the init.jsp when it's loaded several times in the same module, but it seems a replica of the issue we have with the current theme system (in a smaller scale)

It is. Additionally, moving it to the `init.jsp` will probably make it so that we load the CSS for screens that don't really need it. As you can see, adding this manually moves the dial between maintenance and performance. Sometimes one can make that compromise (take a hit on performance or maintenance for the other). I don't think it makes sense in this case since we can probably get the best of both by implementing this differently.

Reply:
The only common tag I can see is `<aui:fieldset>` and `<liferay-frontend:fieldset>` but again, it seems "too much" to me 😞 

![image](https://user-images.githubusercontent.com/46778114/134363886-9e0b3e7e-023a-408f-b4aa-25695f27084d.png)
![image](https://user-images.githubusercontent.com/46778114/134363949-85cd5f93-1930-4bb1-9715-d352478c411d.png)

Reply:
Sorry @edalgrin, forgot to answer back here last week.

> The only common tag I can see is <aui:fieldset> and <liferay-frontend:fieldset> but again, it seems "too much" to me 😞

Maybe it's time to build something then!!

I would check the pattern that uses `.lfr-form-row` or whatever we're trying to style and encapsulate that in a tag or a component. That way, you only need to load the CSS once and all other usages can rely on that rather than having to string together 2 things (a class in html markup and a stylesheet tag).

I don't think this, as it is should move forward. Expecting every developer will know to add a `stylesheet` tag if anywhere in their code they output a `.lfr-form-row` class is unlikely to work.

Reply:
In that case we would be creating a new taglib (that includes the css) for each CSS class in a similar situation, but these classes are mostly legacy and used in a very few cases, so maybe it's not worth it? Another solution (if we wanted to promote the usage of these classe) could be moving them to ClayCSS but, honestly, I don't think we want to do that.

Similar classes (so far):
- `.lfr-asset-column-details`
- `.lfr-form-row`
- `.form-control-inline`
- `.form-search`

You can see a detailed list of it [here](https://docs.google.com/spreadsheets/d/1IG7o3hAn9w91-sg_vXQZeWhJsRBSSMTrGUi-EVfQ6H8/edit#gid=0) (WIP, marked as "Used <liferay-frontend:stylesheet>")

Reply:
> In that case we wouldebe creating a new taglib (that includes the css) for each CSS class in a similar situation, but these classes are mostly legacy and used in a very few cases, so maybe it's not worth it?

It's definitely not worth it, since you're dealing with technical debt here. The really valuable thing to do would be to get rid of those and update the usages and components.

Failing to do that, adding some structure (like a single point of access tag) could be a way to more easily handle that in the future, rather thank sprinkling places that are already legacy with something else that's meant to be removed.

Anyways, those are just my thoughts, you should get some input from the DXP squad and the team to figure out how you want to proceed.